flexmeasures/data/models/time_series.py

```diff
@@ -19,6 +20,93 @@
 class Sensor(db.Model, tb.SensorDBMixin):
     """A sensor measures events. """
 
+    def search_beliefs(
+        self,
+        event_time_window: Tuple[Optional[datetime_type], Optional[datetime_type]] = (
+            None,
+            None,
+        ),
+        belief_time_window: Tuple[Optional[datetime_type], Optional[datetime_type]] = (
+            None,
+            None,
+        ),
+        source: Optional[Union[int, List[int], str, List[str]]] = None,
+    ):
+        """Search all beliefs about events for this sensor.
+
+        :param event_time_window: search only events within this time window
+        :param belief_time_window: search only beliefs within this time window
+        :param source: search only beliefs by this source (pass its name or id) or list of sources"""
+        return TimedBelief.search_all(
+            sensor=self,
+            event_time_window=event_time_window,
+            belief_time_window=belief_time_window,
+            source=source,
+        )
+
+    def __repr__(self) -> str:
+        return f"<Sensor {self.id}: {self.name}>"
+
+
+class TimedBelief(db.Model, tb.TimedBeliefDBMixin):
```

Comment:
In my implementations of `DBTimedBelief`, also a class inheriting from `TimedBeliefDBMixin` as well as an sqlalchemy Base, I chose to write an `TImedBelief.__init__` which explicitly calls both super classes' `__init__` functions. Right now I don't recall what we miss if `db.Model.__init__()` isn't called but I did have a reason which might be reseachable. Did you have a reason not to do that? (Also in the Sensor class actually)

Reply:
Thanks. I had to brush up my knowledge there. I prefer your explicit calls to the superclass `init` functions above relying on the `super` function, and I'll follow your `JoyfulBeliefInCustomTable` example.